charms/worker/k8s/scripts/update_dashboards.py

```diff
@@ -101,17 +101,17 @@ def prepare_dashboard(json_value):
     return json.dumps(json_value, indent=4).replace("$datasource", "$prometheusds")
 
 
-def save_dashboard_to_file(name, data):
+def save_dashboard_to_file(name, data: str):
     """Save the prepared dashboard JSON to a file.
 
     Args:
         name (str): name of the dashboard file
         data (str): file content to write
     """
     filepath = os.path.join(TARGET_DIR, name)
-    with open(filepath, "w") as f:
+    with open(filepath, "w", encoding="utf-8") as f:
```

Comment:
I guess this is a new pylint thing.  whenever you open a file for reading/writing specify its encoding.